### PR TITLE
refactor insights url generator and add annotation support

### PR DESCRIPTION
**Pull request by aarongraham**

### CODE REVIEW

File: lib/ex_twilio/resources/insights/settings.ex [hunk 1]
```diff
@@ -14,4 +14,6 @@ defmodule ExTwilio.Insights.Voice.Settings do
   ]
 
   use ExTwilio.Resource, import: [:find, :create]
+
+  def resource_name, do: "Voice/Settings"
```
Comment (brettbeatty) [inline lib/ex_twilio/resources/insights/settings.ex:18]: What's the difference between a compound resource_name like this and what's going on in Annotation, where the Voice parent is defined?

Comment (aarongraham) [reply lib/ex_twilio/resources/insights/settings.ex:18]: I think doing with the parent is how it is supposed to be done. The problem is as is documented [here](https://github.com/podium/ex_twilio/blob/c09cb580402dc21744e4f73f799aab65b43904d7/lib/ex_twilio/resource.ex#L104) there is always expected to be an identifier after a parent. Which kind of makes sense from a rest pov, but it isn't how this endpoint works strangely.

Comment (aarongraham) [reply lib/ex_twilio/resources/insights/settings.ex:18]: This is really the root of the question I mentioned I had earlier, I'm not sure if I am doing this wrong or if the insights endpoints just behave a bit differently

Comment (brettbeatty) [reply lib/ex_twilio/resources/insights/settings.ex:18]: Ah okay I think I understand now.

Yeah this doesn't seem ideal, but I think it's as good as we'll get without reworking stuff.

File: lib/ex_twilio/resources/insights/settings.ex [hunk 2]
```diff
@@ -1,4 +1,4 @@
-defmodule ExTwilio.Insights.Voice.Settings do
+defmodule ExTwilio.Insights.Settings do
```
Comment (brettbeatty) [inline lib/ex_twilio/resources/insights/settings.ex:1]: If we were doing a versioned library I'd be wary of changing an existing module like this, but I think this is fine for now

**Review by brettbeatty — APPROVED** [review]

**Pull request merged**